server/lib/data_import/my_anime_list/extractor/media.rb

```diff
@@ -0,0 +1,168 @@
+module DataImport
+  class MyAnimeList
+    module Extractor
+      class Media
+        attr_reader :data
+
+        def initialize(json)
+          @data = JSON.parse(json)
+        end
+
+        def age_rating
+          return unless data['classification']
+          rating = data['classification'].split(' - ')
+
+          case rating[0]
+          when 'G', 'TV-Y7' then :G
+          when 'PG', 'PG13' then :PG
+          when 'R', 'R+' then :R
+          when 'Rx' then :R18
+          end
+        end
+
+        def episode_count
+          data['episodes']
+        end
+
+        def episode_length
+          data['duration']
+        end
+
+        def synopsis
+          clean_desc(data['synopsis'])
+        end
+
+        def youtube_video_id
+          data['preview']&.split('/')&.last
+        end
+
+        def poster_image
+          data['image_url']
+        end
+
+        def age_rating_guide
+          return unless data['classification']
+          rating = data['classification'].split(' - ')
+
+          return 'Violence, Profanity' if rating[0] == 'R'
+          return rating[1] if rating[1].present?
+
+          # fallback
+          case rating[0]
+          when 'G' then 'All Ages'
+          when 'PG' then 'Children'
+          when 'PG13', 'PG-13' then 'Teens 13 or older'
+          # when 'R' then 'Violence, Profanity'
+          # this will NEVER happen because of return
+          when 'R+' then 'Mild Nudity'
+          when 'Rx' then 'Hentai'
+          end
+        end
+
+        def subtype # will be renamed to this hopefully
+          # anime matches [TV special OVA ONA movie music]
+          # manga matches [manga novel manhua oneshot doujin]
+
+          case data['type'].downcase
+          # anime
+          when 'tv' then :TV
+          when 'special' then :special
+          when 'ova' then :OVA
+          when 'ona' then :ONA
+          when 'movie' then :movie
+          when 'music' then :music
+          # manga
+          when 'manga' then :manga
+          when 'novel' then :novel
+          when 'manuha' then :manuha
```

Comment:
It's `manhua` not `manuha`.  Also is there a reason why you don't just use `else data['type'].downcase.to_sym` for most of these cases?